core/src/test/java/org/apache/iceberg/CachingCatalogTestHelper.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+import com.github.benmanes.caffeine.cache.Ticker;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.apache.iceberg.catalog.Catalog;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.junit.Assert;
+
+/**
+ * Utility class for accessing package-private VisibleForTesting annotated methods
+ * outside the org.apache.iceberg package.
+ */
+public class CachingCatalogTestHelper {
+
+  // TODO - Consider making this a delegator?
+  private CachingCatalogTestHelper() {
+  }
+
+   // Escape package-private with VisibleForTesting annotation
+  public static CachingCatalog wrapWithTicker(Catalog catalog, boolean expirationEnabled, long expirationIntervalMillis,
+      Ticker ticker) {
+    return (CachingCatalog) CachingCatalog.wrap(catalog, expirationEnabled, expirationIntervalMillis, ticker);
+  }
+
+  public static void assertTableIsCached(String assertionMessage, CachingCatalog catalog, TableIdentifier identifier) {
+    catalog.cache().cleanUp();
+    Assert.assertTrue(assertionMessage, catalog.tableFromCacheQuietly(identifier).isPresent());
+  }
+
+  public static void assertCatalogHasExpiredTable(CachingCatalog catalog, TableIdentifier tableIdent) {
+    Assert.assertFalse("The catalog should not serve table's that are past their TTL",
```

Comment:
maybe replace this with `Assertions.assertThat(catalog.cache().asMap()).doesNotContainKey(tableIdent);`. Then you don't need to have a specific error message, since this code will actually show you the entries in the cache when the test fails, thus it will be quite obvious why this check failed

Reply:
That's a great suggestion. Thank you. The assertion messages in this case aren't really adding much.

After I added that `CachingCatalogTestHelper` class last minute, things got admittedly a little ugly. Appreciate the input!